addons/website/models/ir_rule.py

```diff
@@ -9,14 +9,15 @@ class IrRule(models.Model):
     @api.model
     def _eval_context(self):
         res = super(IrRule, self)._eval_context()
-
         # We need is_frontend to avoid showing website's company items in backend
         # (that could be different than current company). We can't use
         # `get_current_website(falback=False)` as it could also return a website
         # in backend (if domain set & match)..
         is_frontend = ir_http.get_request_website()
+        # Do not provide the current website in context if the rule is about websites.
+        is_not_on_website = self and self.model_id.model != 'website'
         Website = self.env['website']
-        res['website'] = is_frontend and Website.get_current_website() or Website
+        res['website'] = (is_frontend and is_not_on_website) and Website.get_current_website() or Website
```

Comment:
We can probably just remove the `website` context entry from the eval context completely when we are inside the `website` model, no? At least that's what I had in mind, not forcing the record to an empty recordset
At least in master, here maybe it's safer in case someone wrote a weird rule using the eval context `website` record on a `website` record rule. That's very unlikely but could be worth considering it with a TODO to explain it and that it's to be changed in master

Reply:
I just replaced all this with a `sudo` as suggested by xmo [here](https://github.com/odoo/odoo/pull/162892#pullrequestreview-2016880895). I think this will have the minimal impact.